pkg/identity/member.go

```diff
@@ -0,0 +1,78 @@
+package identity
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"time"
+)
+
+// A MemberNodeIdentity identifies a node participating in a community, belonging to a user.
+// It consists of a certificate signed by the user's private key, along with a corresponding private key.
+type MemberNodeIdentity struct {
+	NodeID  string
+	Cert    *x509.Certificate
+	PrivKey *rsa.PrivateKey
+
+	CertBytes       []byte
+	PrivateKeyBytes []byte
```

Comment:
instead of storing the private key directly here should we instead store the path and then read it at time of use? 

also why is there both `Cert` and `CertBytes`? at first read this is confusing since it seems duplicated

Reply:
I added both because different APIs require either the bytes, or the cert. The way go treats x509 is a big mess in my opinion. For the file stuff - I actually didn't implement member node certificate writing yet, going to do that later on when we actually need it

Reply:
ah yikes ok! could also be nice to have a function that converts between the two instead. fine with either